### PR TITLE
Analyzing the dependencies

### LINKED ISSUE #190: Design a "CogniCrypt properties" menu
Some CogniCrypt features may be configurable by its user. Create a menu that allows for this configuration.
It should at least contain the following options:
- [x] Source of CrySL rules (drop-down menu with 'CogniCrypt-internal rules' being pre-selected)
- [x] En-/Disable automated analysis when saving
- [ ] En-/Disable automatic analysis of dependencies
- [ ] En-/Disable automatic analysis of dependencies on change (gets auto-selected when previous option is selected)

- [x] Advanced options
  - [x] Selection of call-graph construction algorithm
  - [x] ~~Selection of entry point (is only available when CHA is not selected as CG algorithm)~~
  - [x] Selection of error-marker types for each error type

### PR DESCRIPTION
**Pull request by shahrzadav**
# Description

New context menu button for analyzing dependencies of project added, also added one more option in preference page for including dependencies while analyzing the project.

Fixes #190

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

It has been tested in eclipse runtime with maven and non maven projects.

**Test Configuration**:
* Eclipse Version: 2018-12 (4.10.0)
* Java Version: 1.8
* OS: Windows 10

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by kruegers — CHANGES_REQUESTED** [review]
Thanks for the pull request. Before I review it, please cleanup your source files first.  
1) Methods should start with a lowercase letters.
2) Comented code should either be removed or uncommented (depending on whether it is needed or not.)
3) Implement the usual error logging through the Activator class and a meanigful error message.

**Comment by shahrzadav** [comment]
@kruegers sorry for delay, would you check please? I made a few changes as you requested

File: plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/handlers/RunAnalysisHandler.java [hunk 1]
```diff
@@ -14,13 +24,52 @@
...
 		IProject ip = Utils.getCurrentlySelectedIProject();
+		
+//		check if the open editor is in the same project
+		if (ip == Utils.getCurrentProject()) {
+			try {
+//				check if there are unsaved changes
+				if (Utils.getCurrentlyOpenEditor().isDirty()) {
+					int answr = saveFile(Utils.getCurrentlyOpenFile());
+//					save file and analyse
+					if (answr == JOptionPane.NO_OPTION) {
```
Comment (kruegers) [inline plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/handlers/RunAnalysisHandler.java:35]: Superfluous.

File: plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/handlers/RunAnalysisHandler.java [hunk 2]
```diff
@@ -14,13 +24,52 @@
...
+						Utils.getCurrentlyOpenEditor().doSave(null);	
+					}
+//					no analyse no save file
+					else if (answr == JOptionPane.CLOSED_OPTION) {
+						return null;
+					}
+				}
+			}
+			catch (Exception e) {
+				// TODO Auto-generated catch block
```
Comment (kruegers) [inline plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/handlers/RunAnalysisHandler.java:48]: Please implement appropriate exception handling here.

File: plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/handlers/RunAnalysisHandler.java [hunk 3]
```diff
@@ -14,13 +24,52 @@
...
+			}
+		}
 		final IJavaElement iJavaElement = JavaCore.create(ip);
 		if (akf.setUp(iJavaElement)) {
 			akf.run();
 		}
 		return null;
 	}
 
+	public static int saveFile(IFile openFileInEditor) throws Exception {
```
Comment (kruegers) [inline plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/handlers/RunAnalysisHandler.java:59]: This should not be part of this PR, should it? Please use branches when working on separate issues.

File: plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/results/ResultsCCUIListener.java [hunk 4]
```diff
@@ -102,17 +104,25 @@ public IProject getReporterProject() {
...
+		this.depVariable = dependencyAnalyser;
+	}
+
 	@Override
 	public void reportError(final AbstractError error) {
 		final String errorMessage = error.toErrorMarkerString();
 		final Statement errorLocation = error.getErrorLocation();
-		final IResource sourceFile = unitToResource(errorLocation);
+		IResource sourceFile = null;
+		if (this.depVariable) {
```
Comment (kruegers) [inline plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/results/ResultsCCUIListener.java:116]: Please move this block to first line of this method.

File: plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/sootbridge/SootRunner.java [hunk 5]
```diff
@@ -203,27 +220,30 @@ private static String getSootClasspath(final IJavaProject javaProject) {
...
+		
 		Collection<String> libraryClassPath = Sets.newHashSet();
 		IClasspathEntry[] rentries;
 		try {
-			rentries = project.getRawClasspath();
-			for (IClasspathEntry entry : rentries) {
-				resolveClassPathEntry(entry, libraryClassPath, project);
-			}
+			// check if "include dependencies" checkbox is checked in preference page or analysis is running for dependencies 
+			if (store.getBoolean(Constants.ANALYSE_DEPENDENCIES) == true || dependencyAnalyser == true) {
```
Comment (kruegers) [inline plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/sootbridge/SootRunner.java:230]: Don't make statements longer than necessary:
a == true <==> a

**Review by kruegers — CHANGES_REQUESTED** [review]

**Pull request closed without merging**